docs/source/common/lightning_cli.rst

```diff
@@ -260,6 +264,62 @@ A possible config file could be as follows:
 Only model classes that are a subclass of :code:`MyModelBaseClass` would be allowed, and similarly only subclasses of
 :code:`MyDataModuleBaseClass`.
 
+Note that with the subclass modes the :code:`--help` option does not show information for a specific subclass. To get
+help for a subclass the options :code:`--model.help` and :code:`--data.help` can be used, followed by the desired class
+path. Similarly the single :code:`--print_config` option does not include the settings for a particular subclass. To
+include them the class path should be given before :code:`--print_config` option. Examples for both help and print
+config are:
+
+.. code-block:: bash
+
+    $ python trainer.py --model.help mycode.mymodels.MyModel
+    $ python trainer.py --model mycode.mymodels.MyModel --print_config
+
+
+Models with multiple submodules
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Many use cases require to have several modules each with its own configurable options. One possible way to handle this
+with LightningCLI is to implement a single module having as init parameters each of the submodules. Since the init
+parameters have as type a class, then in the configuration these would be specified with :code:`class_path` and
+:code:`init_args` entries. For instance a model could be implemented as:
+
+.. testcode::
+
+    class MyMainModel(LightningModule):
+
+        def __init__(
+            self,
+            encoder: EncoderBaseClass,
+            decoder: DecoderBaseClass
+        ):
+            """Example encoder-decoder submodules model
+
+            Args:
+                encoder: Instance of a module for encoding
+                decoder: Instance of a module for decoding
+            """
+            super().__init__()
+            self.encoder = encoder
+            self.decoder = decoder
+
+If the cli is implemented as :code:`LightningCLI(MyMainModel)` the configuration would be as follows:
+
+.. code-block:: yaml
+
+    model:
+      encoder:
+        class_path: mycode.myencoders.MyEncoder
+        init_args:
+          ...
+      decoder:
+        class_path: mycode.mydecoders.MyDecoder
+        init_args:
+          ...
+
+It is also possible to combine :code:`subclass_mode_model=True` and submodules, thereby having two levels of
+:code:`class_path`.
```

Comment:
we should use this more across our docs :)

Reply:
Is this different to ``` ``class_path`` ```?

Reply:
It is the same. Just that inside the model's `init_args` there is a parameter (submodule) defined another `class_path` and `init_args`.